Assignment 3.Rmd

```diff
@@ -138,11 +149,45 @@ What patterns do you see in the plot?
 It would be useful to determine how many people are in each cluster. We can do this easily with dplyr.
 
 ```{r}
-K7 <- 
+K7 <- K4 %>% group_by (cluster) %>% summarise("count" = n())
 ```
 
 Look at the number of people in each cluster, now repeat this process for 3 rather than 2 clusters. Which cluster grouping do you think is more informative? Write your answer below:
 
+```{r}
+set.seed(4050)
+
+fit_3 <- kmeans(K3, 3)
+
+fit_3$cluster
+
+K4_3 <- data.frame(K3, fit_3$cluster)
+
+names(K4_3) <- c("1", "2", "3", "4", "5", "cluster")
+
+K5_3 <- K4_3 %>% gather(key = week,
+                        value = motivation_score,
+                        c(1, 2, 3, 4, 5)
+                        )
+
+K6_3 <- K5_3 %>% group_by (cluster, week) %>% summarise("mean_motivation" = mean(motivation_score))
+
+K6_3$week <- as.numeric(K6_3$week)
+
+K6_3$cluster <- as.factor(K6_3$cluster)
+
+ggplot(K6_3, aes(x = week, y = mean_motivation, colour = cluster)) +
+  geom_line() +
+  labs(x = "Week",
+       y = "Average Motivation",
+       title = "A Graph of Mean Motivation by Week of Two Clusters in the Class"
+       )
+```
+```{r}
+K7 <- K4_3 %>% group_by(cluster) %>% summarise("count" = n())
+```
+
+
```

Comment:
Great job on the coding aspect of Part I! You are on the right track but try to complete Parts II and III.